src/renderer/app-update-warning/app-update-warning.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { makeObservable, observable } from "mobx";
+import { UpdateAvailableChannel } from "../../common/ipc";
+import { Singleton } from "../utils";
+import type { IpcRenderer } from "electron";
+
+interface Dependencies {
+  readonly ipcRenderer: IpcRenderer;
+  readonly sessionStorage: Storage;
+}
+
+const onceADay = 1000 * 60 * 60 * 24;
+
+export class AppUpdateWarning extends Singleton {
```

Comment:
Hmm. Baseclass `Singleton` should not be used for any new code, or at all. I thought it was deprecated, will check again.